src/core/GizmoHelper.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react'
+import { createPortal, useFrame, useThree } from 'react-three-fiber'
+import { Group, Intersection, Matrix4, Object3D, Quaternion, Raycaster, Scene, Vector3 } from 'three'
+import { OrthographicCamera } from './OrthographicCamera'
+import { useCamera } from './useCamera'
+
+type GizmoHelperContext = {
+  tweenCamera: (direction: Vector3) => void
+  raycast: (raycaster: Raycaster, intersects: Intersection[]) => void
+}
+
+const Context = React.createContext<GizmoHelperContext>({} as GizmoHelperContext)
+
+export const useGizmoHelper = () => {
+  return React.useContext<GizmoHelperContext>(Context)
+}
+
+const turnRate = 2 * Math.PI // turn rate in angles per second
+const dummy = new Object3D()
+const matrix = new Matrix4()
+const [q1, q2] = [new Quaternion(), new Quaternion()]
+const target = new Vector3()
+const targetPosition = new Vector3()
+const targetQuaternion = new Quaternion()
+
+export type GizmoHelperProps = JSX.IntrinsicElements['group'] & {
+  alignment?: 'top-left' | 'top-right' | 'bottom-right' | 'bottom-left'
+  margin?: [number, number]
+  onUpdate: () => void // update controls during animation
+  onTarget: () => Vector3 // return the target to rotate around
+}
+
+export const GizmoHelper = ({
+  alignment = 'bottom-right',
+  margin = [80, 80],
+  onUpdate,
+  onTarget,
+  children: GizmoHelperComponent,
+}: GizmoHelperProps): any => {
+  const { gl, scene, camera: mainCamera, size } = useThree()
+  const gizmoRef = React.useRef<Group>()
+  const virtualCam = React.useRef<any>()
+  const virtualScene = React.useMemo(() => new Scene(), [])
+  const [animating, setAnimating] = React.useState(false)
```

Comment:
i'm concerned about using this many `useStates` why can't animating be a `ref`? same for the ones below....